core/Cargo.toml

```diff
@@ -56,9 +67,10 @@ rand07 = { package = "rand", version = "0.7" }
 prost-build = "0.9"
 
 [features]
-default = [ "secp256k1", "ecdsa" ]
-secp256k1 = [ "libsecp256k1" ]
-ecdsa = [ "p256" ]
+default = ["secp256k1", "ecdsa"]
+secp256k1 = ["libsecp256k1"]
+ecdsa = ["p256"]
+use-serde = ["multihash/serde-codec", "serde"]
```

Comment:
We can avoid the awkward `use-serde` name by renaming the `serde` dependency like this:

```toml
serde_crate = { package = "serde", version = "1" }
```

and then do:

```rust
extern crate serde_crate as serde;
```

Inside `lib.rs`.

Reply:
Also here, please undo the unnecessary re-formatting.

Reply:
Referencing https://github.com/rust-lang/api-guidelines/discussions/180 here as a general Rust discussion. If I understand correctly, ideally we could use `serde` for both feature and crate, though that seems [not to be stable yet](https://github.com/rust-lang/cargo/issues/5565#issuecomment-412296421).

I don't have an opinion. I would go with whatever is most widely used within the Rust ecosystem.

Reply:
I did this b/c I've seen a `use-serde` feature on other crates. I tried @thomaseizinger's solution, but it seems that `serde` doesn't support being aliased, because when I tried aliasing `serde` as `serde_crate` or `_serde`, then I started getting this error on the derive macros:

```plain
error[E0463]: can't find crate for `serde`
  --> core/src/peer_id.rs:37:38
   |
37 | #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
   |                                      ^^^^^^^^^ can't find crate
   |
   = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)
```